cluster-autoscaler/cloudprovider/gridscale/gridscale_node_group.go

```diff
@@ -0,0 +1,413 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gridscale
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gridscale/gsclient-go/v3"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/framework"
+)
+
+var (
+	// ErrNodePoolNotExist is return if no node pool exists for a given cluster ID
+	ErrNodePoolNotExist = errors.New("node pool does not exist")
+)
+
+// Interface guard
+var _ cloudprovider.NodeGroup = (*NodeGroup)(nil)
+
+// NodeGroup implements cloudprovider.NodeGroup interface. NodeGroup contains
+// configuration info and functions to control a set of nodes that have the
+// same capacity and set of labels.
+type NodeGroup struct {
+	id          string
+	name        string
+	clusterUUID string
+	client      nodeGroupClient
+	nodeCount   int
+	minSize     int
+	maxSize     int
+}
+
+// MaxSize returns maximum size of the node group.
+func (n *NodeGroup) MaxSize() int {
+	return n.maxSize
+}
+
+// MinSize returns minimum size of the node group.
+func (n *NodeGroup) MinSize() int {
+	return n.minSize
```

Comment:
What would happen if we did this?


```suggestion
	if n.minSize == 0 {
		return 1
	} else {
		return n.minSize
	}
```

Would this be a working, good way to ensure the first node of each pool is never scaled down?

Reply:
idea discarded in favor of others.